### PR TITLE
update token

### PR DESCRIPTION
**Pull request by ermalkaleci**
update token to v4.5.0

### CODE REVIEW

File: contracts/token/Token.sol [hunk 1]
```diff
@@ -154,6 +154,7 @@ contract Token is IERC20 {
     function _transfer(address sender, address recipient, uint256 amount) internal {
         require(sender != address(0), "ERC20: transfer from the zero address");
         require(recipient != address(0), "ERC20: transfer to the zero address");
+        require(amount <= type(uint128).max, "ERC20: transfer amount exceeds uint128::max");
```
Comment (zjb0807) [inline contracts/token/Token.sol:157]: The precision of ACA is 12, which is 18 in the contract, so it cannot satisfy u128::max.
Although u128::max is unlikely to happen.

Comment (ermalkaleci) [reply contracts/token/Token.sol]: @zjb0807 in the current design you can pass large number and precompile will ignore first 32 bytes and resulting into a successful transaction but the amount is not the one passed

Comment (ermalkaleci) [reply contracts/token/Token.sol]: I removed this and will handle it on input hander

**Review by zjb0807 — APPROVED** [review]
LGTM

**Comment by zjb0807** [comment]
Still need to update `"@openzeppelin/contracts"` in package.json

**Review by zjb0807 — APPROVED** [review]

**Pull request merged**